src/Symfony/Component/Config/Tests/Definition/Builder/NodeFinderTest.php

```diff
@@ -0,0 +1,114 @@
+        $nodeFinder->find('child', $rootNode);
+    }
+
+    public function testItShouldThrowExceptionIfNodeDoesNotExistInArrayRootNode()
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Node with name "child" does not exist in the given root node "root"!');
+
+        $rootNode = new ArrayNodeDefinition('root');
+        $rootNode
+            ->children()
+                ->arrayNode('social_media_channels')->end()
+            ->end()
+        ;
+
+        $nodeFinder = new NodeFinder();
+        $nodeFinder->find('child', $rootNode);
+    }
+
+    public function testItShouldHandleComplexConfigurationProbably()
+    {
+        $rootNode = new ArrayNodeDefinition('root');
+        $rootNode
+            ->children()
+                ->arrayNode('social_media_channels')
+                    ->children()
+                        ->booleanNode('enable')->end()
+                        ->arrayNode('twitter')->end()
+                        ->arrayNode('facebook')->end()
+                        ->arrayNode('instagram')
+                            ->children()
+                                ->booleanNode('enable')->end()
+                                ->arrayNode('accounts')->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+
+        $nodeFinder = new NodeFinder();
+
+        $socialMediaChannelsNode = $nodeFinder->find('social_media_channels', $rootNode);
+        $socialMediaChannelsEnableNode = $nodeFinder->find('social_media_channels.enable', $rootNode);
+
+        $twitterNode = $nodeFinder->find('social_media_channels.twitter', $rootNode);
+        $facebookNode = $nodeFinder->find('social_media_channels.facebook', $rootNode);
+        $instagramNode = $nodeFinder->find('social_media_channels.instagram', $rootNode);
+        $instagramEnableNode = $nodeFinder->find('social_media_channels.instagram.enable', $rootNode);
+        $instagramAccountsNode = $nodeFinder->find('social_media_channels.instagram.accounts', $rootNode);
+
+        $this->assertNode($socialMediaChannelsNode, 'social_media_channels', ArrayNodeDefinition::class);
+        $this->assertNode($socialMediaChannelsEnableNode, 'enable', BooleanNodeDefinition::class);
+
+        $this->assertNode($twitterNode, 'twitter', ArrayNodeDefinition::class);
+        $this->assertNode($facebookNode, 'facebook', ArrayNodeDefinition::class);
+        $this->assertNode($instagramNode, 'instagram', ArrayNodeDefinition::class);
+        $this->assertNode($instagramEnableNode, 'enable', BooleanNodeDefinition::class);
+        $this->assertNode($instagramAccountsNode, 'accounts', ArrayNodeDefinition::class);
+    }
+
+    /**
+     * @param NodeDefinition $actualNode
+     * @param string         $expectedName
+     * @param string         $expectedType
+     */
+    private function assertNode(NodeDefinition $actualNode, $expectedName, $expectedType)
```

Comment:
Please add string type hint and remove the entire doc bloc

Reply:
Done